udata_schema_gouvfr/views.py

```diff
@@ -5,21 +5,52 @@
 from udata import theme
 from udata.frontend import template_hook
 
+import requests
+
 blueprint = Blueprint('schema', __name__, template_folder='templates')
 
 
-def validata_url(resource):
+def validata_url(resource, schema_url=None):
     base = current_app.config.get('SCHEMA_GOUVFR_VALIDATA_URL')
-
-    query = urlencode({
-        "input": "url",
-        "schema_name": f"schema-datagouvfr.{resource.schema}",
-        "url": resource.url,
-    })
+    if schema_url:
+        query = urlencode({
+            'input': 'url',
+            'schema_url': schema_url,
+            'url': resource.url,
+        })
+    else:
+        query = urlencode({
+            'input': 'url',
+            'schema_name': f"schema-datagouvfr.{resource.schema['name']}",
+            'url': resource.url,
+        })
 
     return f"{base}/table-schema?{query}"
 
 
+def is_table_schema(schemas, current_schema):
+    if schemas:
+        for schema in schemas:
+            if schema['name'] == current_schema and schema['schema_type'] == 'tableschema':
+                return True
+    return False
+
+
+def get_schema_url(schemas, current_schema, current_schema_version):
+    if schemas:
+        for schema in schemas:
+            if schema['name'] == current_schema:
+                for version in schema['versions']:
+                    if version['version_name'] == current_schema_version:
+                        return version['schema_url']
+
+
+def load_catalog():
+    r = requests.get(current_app.config.get('SCHEMA_CATALOG_URL'))
```

Comment:
```suggestion
    r = requests.get(current_app.config.get('SCHEMA_CATALOG_URL'))
    r.raise_for_status()
```

(and maybe handle the exception as you see fit)